packages/snaps-simulator/src/features/builder/components/NodeTree.tsx

```diff
@@ -92,17 +98,20 @@ export const NodeTree: FunctionComponent<NodeTreeProps> = ({
     return false;
   };
 
-  const handleCanDrop = (
-    _tree: NodeModel<Component>[],
-    {
-      dropTarget,
-      dropTargetId,
-    }: {
-      dropTarget?: NodeModel<Component> | undefined;
-      dropTargetId?: string | number;
-    },
+  const handleCanDrop: TreeProps<Component>['canDrop'] = (
+    _tree,
+    { dropTarget, dropTargetId, dragSource },
   ) => {
     if (dropTargetId) {
+      // Checks if the component is allowed in an Form.
+      if (
+        dropTarget?.data?.type === NodeType.Form &&
+        dragSource?.data?.type !== NodeType.Button &&
+        dragSource?.data?.type !== NodeType.Input
+      ) {
+        return false;
+      }
```

Comment:
We should probably come up with a more generic solution in the future, but this will do for now.

Reply:
Yeah, probably a shared array in sdk or utils would save us from doing this in multiple places.